servicetalk-client-api/src/main/java/io/servicetalk/client/api/ConnectionRejectedException.java

```diff
@@ -17,11 +17,13 @@
 
 import io.servicetalk.transport.api.RetryableException;
 
+import java.net.SocketException;
+
 /**
  * Thrown when a newly created connection was rejected.
  */
-public final class ConnectionRejectedException extends RuntimeException implements RetryableException {
-    private static final long serialVersionUID = -2573006397795678870L;
+public class ConnectionRejectedException extends SocketException implements RetryableException {
```

Comment:
Debated between `ConnectException` and `SocketException`. Based on their javadoc, decided to go with `SocketException` because of technically the connection was already established, but there was a problem "accessing" it due to a selector or a host becoming unavailable.